### PR TITLE
jgraph: update to version 8.5

### PR DESCRIPTION
**Pull request by stevecheckoway**
#### Description

This
- changes the license to correctly reflect the actual 3-clause BSD
  license rather than GPL;
- removes me as a maintainer as I haven't used this program in many
  years and have no more interest in it;
- removes the architecture limitation as the code is ANSI C with no
  architectural limitations; and
- removes the patch files (which may be where the GPL license came from)
  as the author of jgraph updated the code to allow it to be compiled
  with newer compilers.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G6042
Xcode 11.3.1 11C505

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by stevecheckoway** [comment]
I'm not entirely sure what the failures are here. I'm happy to fix the errors, but I'd appreciate some insight into what is failing. Something wrong with the maintainer field and lack of tests?

File: graphics/jgraph/Portfile [hunk 1]
```diff
@@ -3,13 +3,13 @@
 PortSystem          1.0
 
 name                jgraph
-version             8.3
+version             8.5
 categories          graphics
 # see http://packages.debian.org/changelogs/pool/main/j/jgraph/current/copyright
-license             GPL
+license             BSD-3-Clause
```
Comment (reneeotten) [inline graphics/jgraph/Portfile:9]: for MacPorts the valid license description here is `BSD` which includes both the 2- and 3-clause

Comment (reneeotten) [inline graphics/jgraph/Portfile:9]:
```suggestion
license             BSD
```

File: graphics/jgraph/Portfile [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
 homepage \
     http://www.cs.utk.edu/~plank/plank/jgraph/jgraph.html
```
Comment (reneeotten) [inline graphics/jgraph/Portfile:11]:
```suggestion
    https://www.cs.utk.edu/~plank/plank/jgraph/jgraph.html
```

Comment (stevecheckoway) [reply graphics/jgraph/Portfile:11]: The site serves a bad cert, hence the http link.

Comment (reneeotten) [reply graphics/jgraph/Portfile:11]: sorry, you're right!

**Comment by reneeotten** [comment]
@stevecheckoway there are some issues with ```error: implicit declaration of function XXXXX```, see the CI logs. Is that something you are able to fix?

**Comment by stevecheckoway** [comment]
I have no idea why it's getting warnings that are then turned into errors. I get no warnings or errors when I run `sudo port build` in the directory. I pushed a change that remove `-pedantic` and replaces it with `-Wno-error`. Let's see how the CI likes that. It seems like clang is ignoring the `-ansi` flag.

Perhaps a better option would be to set the standard via `-std=c89` (which also works fine for me). After this build finishes, I'll check that and if that works, then we can go with that. Once something works, I can squash these commits.

**Comment by reneeotten** [comment]
> I have no idea why it's getting warnings that are then turned into errors. I get no warnings or errors when I run `sudo port build` in the directory. I pushed a change that remove `-pedantic` and replaces it with `-Wno-error`. Let's see how the CI likes that. It seems like clang is ignoring the `-ansi` flag.
> 
> Perhaps a better option would be to set the standard via `-std=c89` (which also works fine for me). After this build finishes, I'll check that and if that works, then we can go with that. Once something works, I can squash these commits.

It's what happens with Xcode 12 and above; see the [thread](https://lists.macports.org/pipermail/macports-dev/2020-November/042648.html) on the mailinglist where @ryandesign explains it.

**Comment by stevecheckoway** [comment]
I added a single patch file that creates headers with the declarations and includes them in the appropriate files. This builds with `-Werror=implicit-function-declaration`.

**Comment by reneeotten** [comment]
> I added a single patch file that creates headers with the declarations and includes them in the appropriate files. This builds with `-Werror=implicit-function-declaration`.

thank you very much - merging this now!

**Comment by stevecheckoway** [comment]
Great, thanks!

**Pull request merged**